### DISCUSSION TITLE
Interface usage

### DISCUSSION THREAD

**Discussion by daohu527** [eProsima/Fast-DDS#5128] [General]
I want to use `Publisher` in `src/cpp/fastrtps_deprecated/publisher`, but I notice the `fastrtps_deprecated`. My question are

- Are these interfaces still usable?
- what interface should I use to replace this interface since they are deprecated？

I created a Participant as follows, then used the Publisher write interface to send data and the Subscriber to receive data. However, I cannot receive data. Can you give some suggestions, especially the configuration of Participant?
https://github.com/wheelos/core/blob/86ecf33d2b37504418c0536b9e56da6190e18cff/cyber/transport/rtps/participant.cc#L64-L140

**Comment by JesusPoderoso** [comment]
Hi @daohu527, thanks for using Fast DDS, and sorry for the inconvenience.
There are many API breaks entailed due to the near Fast DDS v3 release.

> - Are these interfaces still usable?

Some of them have been moved to private API, and some new API to mange them has been created.

> - what interface should I use to replace this interface since they are deprecated？

A [migration guide](https://github.com/eProsima/Fast-DDS/blob/master/UPGRADING.md) has been included for the release to help with the API transition. You can also refer to the refactored [RTPS example](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/rtps) to have a better overview of the changes introduced.

**Comment by daohu527** [comment]
Thanks for your reply, I found the parameters of the participants not explained clearly.

I mean, if we could write documentation about parameters, it would be a disaster for a newbie to set these parameters.

**Comment by JesusPoderoso** [comment]
The [`RTPSDomain::createParticipant` method](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/RTPSDomain.html#_CPPv4N8eprosima7fastdds4rtps10RTPSDomain17createParticipantE8uint32_tRK25RTPSParticipantAttributesP23RTPSParticipantListener) requires [`RTPSParticipantAttributes`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/attributes/RTPSParticipantAttributes.html#_CPPv4N8eprosima7fastdds4rtps25RTPSParticipantAttributesE), which are configured likewise the `fastrtps::ParticipantAttributes` configured in your code snippet.
